platform-hub-api/app/controllers/users_controller.rb

```diff
@@ -43,14 +43,51 @@ def revoke_admin
       action: 'revoke_admin',
       auditable: @user
     )
-    
+
     head :no_content
   end
 
+  # POST /users/:id/onboard_github
+  def onboard_github
+    handle_github_agent_request :onboard
+  end
+
+  # POST /users/:id/offboard_github
+  def offboard_github
+    handle_github_agent_request :offboard
+  end
+
   private
 
   def find_user
     @user = User.find params[:id]
   end
 
+  def handle_github_agent_request agent_action
+    begin
+      success = gitHubAgentService.send "#{agent_action}_user", @user
+
+      if success
+        AuditService.log(
+          context: audit_context,
+          action: action_name,
+          auditable: @user,
+        )
+        head :no_content
+      else
+        render_error "Failed to Github #{agent_action} the user - the GitHub API may be down", :service_unavailable
+      end
+    rescue Agents::GitHubAgentService::Errors::IdentityMissing
+      render_error 'User does not have a GitHub identity connected yet', :bad_request
+    end
+  end
+
+  def gitHubAgentService
```

Comment:
I meant maybe we should have an `agents` initializer in `config/initializers` where we keep those?